src/main/java/uk/gov/homeoffice/digital/sas/balancecalculator/kafka/consumer/TimeEntryConsumer.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.homeoffice.digital.sas.balancecalculator.kafka.consumer;
+
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_COULD_NOT_DESERIALIZE_RESOURCE;
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.KAFKA_RESOURCE_NOT_UNDERSTOOD;
+import static uk.gov.homeoffice.digital.sas.kafka.consumer.KafkaConsumerUtils.getResourceFromMessageAsString;
+import static uk.gov.homeoffice.digital.sas.kafka.consumer.KafkaConsumerUtils.getSchemaFromMessageAsString;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Import;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+import org.springframework.util.ObjectUtils;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.TimeEntry;
+import uk.gov.homeoffice.digital.sas.kafka.consumer.KafkaConsumerService;
+import uk.gov.homeoffice.digital.sas.kafka.consumer.configuration.KafkaConsumerConfig;
+import uk.gov.homeoffice.digital.sas.kafka.exceptions.KafkaConsumerException;
+import uk.gov.homeoffice.digital.sas.kafka.message.KafkaEventMessage;
+
+@Service
+@Slf4j
+@Import(KafkaConsumerConfig.class)
+public class TimeEntryConsumer {
+
+  private ObjectMapper mapper = new ObjectMapper();
+
+  private final KafkaConsumerService<TimeEntry> kafkaConsumerService;
+
+
+  @Autowired
+  public TimeEntryConsumer(KafkaConsumerService<TimeEntry> kafkaConsumerService) {
+    this.kafkaConsumerService = kafkaConsumerService;
+  }
+
+  @KafkaListener(topics = {"${spring.kafka.template.default-topic}"},
+                groupId = "${spring.kafka.consumer.group-id}",
+                errorHandler = "kafkaConsumerErrorHandler")
+  public void onMessage(@Payload String payload) throws JsonProcessingException {
+
+    if (kafkaConsumerService.isResourceOfType(payload, TimeEntry.class)) {
+      KafkaEventMessage<TimeEntry> kafkaEventMessage =
+          kafkaConsumerService.convertToKafkaEventMessage(payload);
+
+      if (!ObjectUtils.isEmpty(kafkaEventMessage)) {
+        TimeEntry timeEntry = createTimeEntryFromKafkaEventMessage(kafkaEventMessage, payload);
+        kafkaConsumerService.checkDeserializedResource(payload, timeEntry);
```

Comment:
The `checkDeserializedResource` function checks if `ObjectUtils.isEmpty(resource)`, so the log for the error when it is empty will never be called from inside this if statement

Reply:
This has been updated in kafka-commons to throw an exception now. Logging an error seemed pointless. This method may become unneeded in the future but for now it is an extra layer of protection when deserializing.

Reply:
Will the exception get hit if it's wrapped in an if block in this file? We already check that the event message isn't empty before calling `checkDeserializedResource`

Reply:
I agree. I can't see a scenario when we would get to this point and timeEntry be empty.
It has been removed from kafka-commons and the success log message moved to the service. (Also added a temporary log message to stop the introduction of a code smell until the calculate logic is implemented).